lib/bolt/plugin/task.rb

```diff
@@ -59,7 +59,7 @@ def puppet_library(opts, target, apply_prep)
         run_opts = {}
         run_opts[:run_as] = opts['_run_as'] if opts['_run_as']
         begin
-          task = apply_prep.get_task(opts['task'], params)
+          task = @context.get_validated_task(opts['task'], params)
```

Comment:
`apply_prep.run_task` also seems like it can just use `@context.run_task`

Reply:
There's no `PluginContext.run_task` method, and I don't think writing one is really in the scope of this ticket.

Reply:
Oh, gosh, sorry I was looking at `Bolt::Task::Run.run_task()`, which we can use here. I don't think it matters that much, it just seems odd to have so many functions that just wrap `executor.run_task`. Reduce reuse recycle, etc.

Reply:
Hrm... to use `Bolt::Task::Run.run_task()` here we'd need to pass the executor to `Bolt::Plugin::Module.puppet_library` and `Bolt::Plugin::Task.puppet_library`. At that point why not just call `executor.run_task` directly?